src/manifolds/ProductManifold.jl

```diff
@@ -183,6 +189,12 @@ function check_vector(
     (length(errors) == 1) && return cerr[1]
     return nothing
 end
+function check_vector(M::ProductManifold, p, X; kwargs...)
+    return DomainError(
+        typeof(X),
+        "The vector $X is not a tangent vector to any tangent space on $M, since currently only ProductRepr and ProductArray are suppored typoes for tangent vectors on product manifolds",
```

Comment:
```suggestion
        "The vector $X is not a tangent vector to any tangent space on $M, since currently only ProductRepr and ProductArray are supported types for tangent vectors on product manifolds",
```